Assets/LeapMotionModules/Interaction/Scripts/InteractionHand.cs

```diff
@@ -8,947 +8,502 @@
         }
       }
+      else { // Otherwise, configure automatically.
+        if (leapProvider == null) {
+          leapProvider = Hands.Provider;
 
-      return null;
-    }
-
-    // Disables broadphase checks if an object is currently interacting with this hand.
-    private bool _interactionHoverOverride = false;
-
-    public struct HoverCheckResults {
-      public HashSet<IInteractionBehaviour> hovered;
-      public IInteractionBehaviour primaryHovered;
-      public int primaryHoveringFingerIdx;
-      public float primaryHoveredDistance;
-      public IInteractionBehaviour[] perFingerHovered;
-      public float[] perFingerDistance;
-      public Hand checkedHand;
-    }
-
-    private HoverCheckResults _hoverResults = new HoverCheckResults();
-
-    /// <summary>
-    /// Returns the information used to calculate hover and primary hover callbacks from
-    /// the latest fixed frame.
-    /// </summary>
-    public HoverCheckResults hoverCheckResults { get { return _hoverResults; } }
-
-    private void FixedUpdateHovering() {
-      using (new ProfilerSample("Fixed Update InteractionHand Hovering")) {
-        if (_hand == null && _interactionHoverOverride) { _interactionHoverOverride = false; }
-
-        if (!_interactionHoverOverride) {
-          hoverActivityManager.activationRadius = interactionManager.WorldHoverActivationRadius;
-          hoverActivityManager.FixedUpdateQueryPosition((_hand != null) ? _hand.PalmPosition.ToVector3() : Vector3.zero, LeapSpace.allEnabled);
-
-          using (new ProfilerSample("Check for Closest Elements")) {
-            CheckHoverForHand(_hand, _hoverActivityManager.ActiveObjects);
+          if (leapProvider == null) {
+            Debug.LogError("No LeapServiceProvider was found in your scene! Please "
+                         + "make sure you have a LeapServiceProvider if you intend to "
+                         + "use Leap hands in your scene.", this);
+            return;
           }
         }
 
-        ProcessHoverCheckResults();
-
-        // Prevents being able to depress two buttons at once by rapidly alternating
-        // primary hover.
-        //if (_primaryHoveredLastFrame == null || !_contactBehaviours.ContainsKey(_primaryHoveredLastFrame)) {
-          ProcessPrimaryHoverCheckResults();
-        //}
-
-        ISpaceComponent space;
-        if (_hand != null) {
-          _warpedHandData.CopyFrom(_handData);
-          if (_hoverResults.primaryHovered != null && (space = _hoverResults.primaryHovered.space) != null) {
-            //Transform bulk hand to the closest element's warped space
-            inverseTransformHand(_warpedHandData, space);
+        if (handAccessorFunc == null) {
+          if (handDataMode == HandDataMode.PlayerLeft) {
+            handAccessorFunc = (frame) => frame.Hands.Query()
+                                                     .Where(hand => hand.IsLeft)
+                                                     .FirstOrDefault();
```

Comment:
just a small note that you can pass predicates directly to First (and many direct ops) instead of prefixing them with a where :)